### ISSUE TITLE
fix(jest): automatic checks should not be triggered for failed tests

### ISSUE THREAD

**Issue by mohanraj-r** [salesforce/sa11y#66]
Automatic check is triggered regardless of the test status which would result in the original test failure if any getting overwritten by a11y failures if any from automatic checks.

### Current
- Enable automatic checks in a parallel CI workflow and not in the main CI workflow.

### Future
- With the default Jasmine test runner in Jest v26 there doesn't seem to be clean way to [detect test failures from Jest's teardown hooks](https://github.com/facebook/jest/issues/5292)
- But [jest-circus](https://github.com/facebook/jest/tree/master/packages/jest-circus) offers event handlers which allows binding to test failure events among others
  - `jest-circus` is the default test runner starting Jest v27. After Jest v27 gains more ground and known issues (#65) are fixed this could be a viable option
